icu4c/source/common/unicode/platform.h

```diff
@@ -347,6 +347,18 @@
 #   define U_HAVE_INTTYPES_H U_HAVE_STDINT_H
 #endif
 
+/**
+ * \def U_SUPPORTS_ATOMICS
+ * Defines whether the platform supports C++ atomics.
+ * Set this to 0 to disable certain code that requires atomics.
```

Comment:
I would add another sentence like
Only a small part of the ICU code currently avoids C++ atomics based on this setting.